feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/start/StartStakingInteractorFactory.kt

```diff
@@ -0,0 +1,78 @@
+import io.novafoundation.nova.feature_staking_impl.domain.era.StakingEraInteractorFactory
+import io.novafoundation.nova.feature_staking_impl.domain.parachainStaking.main.ParachainNetworkInfoInteractor
+import io.novafoundation.nova.feature_staking_impl.domain.parachainStaking.rewards.ParachainStakingRewardCalculatorFactory
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.direct.ParachainStartStakingInteractor
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.direct.RelaychainStartStakingInteractor
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.nomination.NominationPoolStartStakingInteractor
+import io.novafoundation.nova.feature_wallet_api.domain.interfaces.WalletRepository
+import io.novafoundation.nova.runtime.ext.StakingTypeGroup
+import io.novafoundation.nova.runtime.ext.group
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import kotlinx.coroutines.CoroutineScope
+
+class StartStakingInteractorFactory(
+    private val stakingSharedComputation: StakingSharedComputation,
+    private val walletRepository: WalletRepository,
+    private val accountRepository: AccountRepository,
+    private val stakingEraInteractorFactory: StakingEraInteractorFactory,
+    private val parachainNetworkInfoInteractor: ParachainNetworkInfoInteractor,
+    private val parachainStakingRewardCalculatorFactory: ParachainStakingRewardCalculatorFactory
+) {
+
+    suspend fun create(chain: Chain, asset: Chain.Asset, coroutineScope: CoroutineScope): CompoundStartStakingInteractor {
+        val interactors = createInteractors(chain, asset, coroutineScope)
+        val stakingEraInteractor = stakingEraInteractorFactory.create(chain)
+        return RealCompoundStartStakingInteractor(walletRepository, accountRepository, interactors, stakingEraInteractor)
+    }
+
+    private suspend fun createInteractors(chain: Chain, asset: Chain.Asset, coroutineScope: CoroutineScope): List<StartStakingInteractor> {
+        return asset.staking.mapNotNull {
+            when (it.group()) {
+                StakingTypeGroup.RELAYCHAIN -> createRelaychainStartStakingInteractor(coroutineScope)
+                StakingTypeGroup.PARACHAIN -> createPararchainStartStakingInteractor(coroutineScope, chain, asset, it)
+                StakingTypeGroup.NOMINATION_POOL -> createNominationPoolsStartStakingInteractor(coroutineScope)
+                StakingTypeGroup.UNSUPPORTED -> null
+            }
+        }
+    }
+
+    private fun createRelaychainStartStakingInteractor(coroutineScope: CoroutineScope): StartStakingInteractor {
+        return RelaychainStartStakingInteractor(
+            stakingSharedComputation = stakingSharedComputation,
+            accountRepository = accountRepository,
+            walletRepository = walletRepository,
+            coroutineScope = coroutineScope,
+            stakingType = Chain.Asset.StakingType.RELAYCHAIN
```

Comment:
This might not be correct since stakingStype might be RELAYCHAIN_AURA too. Better to pass stakingType directly